homeassistant/components/samsungtv/bridge.py

```diff
@@ -359,6 +353,22 @@ def _get_remote(self, avoid_open: bool = False) -> Remote:
                 self._remote = None
         return self._remote
 
+    def is_on(self) -> bool:
+        """Tells if the TV is on."""
+        if self._remote is not None:
+            self.close_remote()
+        try:
+            info = self.device_info()
+            if info is not None and isinstance(info["device"]["PowerState"], str):
+                return info["device"]["PowerState"] == "on"
+            return False
+        except (UnhandledResponse, AccessDenied, ConnectionFailure, KeyError):
+            # We got a response so it's working.
+            return True
+        except (OSError):
+            # Different reasons, e.g. hostname not resolveable
+            return False
+
```

Comment:
Please move this logic to `update` or `async_update`

Reply:
Since we can't determine if the TV is on or off without these checks, what should `is_on` return in this case?